src/main/java/frc/robot/subsystems/swerve/module/Module.java

```diff
@@ -0,0 +1,260 @@
+	}
+
+	public void setTargetVelocity(double targetVelocityMetersPerSecond, Rotation2d targetSteerPosition, boolean isClosedLoop) {
+		targetVelocityMetersPerSecond = ModuleUtils.reduceSkew(targetVelocityMetersPerSecond, targetSteerPosition, getSteerPosition());
+
+		if (isClosedLoop) {
+			setTargetClosedLoopVelocity(targetVelocityMetersPerSecond);
+		} else {
+			setTargetOpenLoopVelocity(targetVelocityMetersPerSecond);
+		}
+	}
+
+	public void setTargetClosedLoopVelocity(double targetVelocityMetersPerSecond) {
+		setClosedLoop(true);
+		Rotation2d targetVelocityPerSecond = Conversions.distanceToAngle(targetVelocityMetersPerSecond, constants.wheelDiameterMeters());
+		Rotation2d coupledVelocityPerSecond = ModuleUtils
+			.coupleAngle(targetVelocityPerSecond, steerStuff.velocitySignal().getLatestValue(), constants.couplingRatio());
+		drive.applyAngleRequest(driveVelocityRequest.withSetPoint(coupledVelocityPerSecond));
+	}
+
+	public void setTargetOpenLoopVelocity(double targetVelocityMetersPerSecond) {
+		double voltage = ModuleUtils.velocityToOpenLoopVoltage(
+			targetVelocityMetersPerSecond,
+			steerStuff.velocitySignal().getLatestValue(),
+			constants.couplingRatio(),
+			constants.velocityAt12VoltsPerSecond(),
+			constants.wheelDiameterMeters(),
+			ModuleConstants.VOLTAGE_COMPENSATION_SATURATION
+		);
+		setDriveVoltage(voltage);
+	}
+
+
+	/**
+	 * The odometry thread can update itself faster than the main code loop (which is 50 hertz). Instead of using the latest odometry update, the
+	 * accumulated odometry positions since the last loop to get a more accurate position.
+	 *
+	 * @param odometryUpdateIndex the index of the odometry update
+	 * @return the position of the module at the given odometry update index
+	 */
+	public SwerveModulePosition getOdometryPosition(int odometryUpdateIndex) {
+		return new SwerveModulePosition(
+			driveInputs.positionsMeters[odometryUpdateIndex],
+			steerStuff.positionSignal().asArray()[odometryUpdateIndex]
+		);
+	}
+
+	public int getNumberOfOdometrySamples() {
+		return Math.min(driveInputs.positionsMeters.length, steerStuff.positionSignal().asArray().length);
+	}
+
+	public SwerveModuleState getTargetState() {
+		return targetState;
+	}
+
+	public SwerveModuleState getCurrentState() {
+		return new SwerveModuleState(getDriveVelocityMetersPerSecond(), getSteerPosition());
+	}
+
+	public Rotation2d getDriveAngle() {
+		return driveInputs.uncoupledPositions[driveInputs.uncoupledPositions.length - 1];
+	}
+
+	public double getDriveVelocityMetersPerSecond() {
+		return driveInputs.velocityMetersPerSecond;
+	}
+
+	public Rotation2d getSteerPosition() {
+		return steerStuff.positionSignal().getLatestValue();
+	}
+
+
+	//@formatter:off
+	public boolean isAtTargetVelocity(double speedToleranceMetersPerSecond) {
+		return MathUtil.isNear(
+				getTargetState().speedMetersPerSecond,
+				getDriveVelocityMetersPerSecond(),
+				speedToleranceMetersPerSecond
```

Comment:
use a single tab for indentation instead of two
```suggestion
			getTargetState().speedMetersPerSecond,
			getDriveVelocityMetersPerSecond(),
			speedToleranceMetersPerSecond
```
to prevent this from happening in the future:
1. go to settings->Editor->Code Style->Java
2. change "Continuation indent" from its default value of 8 to 4